### PR TITLE
Start testing AArch64 on CI

### PR DESCRIPTION
**Pull request by BertalanD**
This should help us avoid accidentally breaking the build for AArch64.
Currently, some tests are expected to fail, so CI runs will be considered successful even if the kernel panics or if there are test failures.

For now, we have to build Qemu with a custom patch from source in order for SystemServer to detect self-test mode.

### CODE REVIEW

File: Toolchain/BuildIt.sh [hunk 1]
```diff
@@ -372,6 +372,15 @@ popd
 pushd "$DIR"
     if [ "${TRY_USE_LOCAL_TOOLCHAIN}" = "y" ] ; then
         echo "::endgroup::"
+
+        # FIXME: Qemu currently needs a local patch for AArch64 testing.
+        #        It is included in Qemu 8.1; remove this when upgrading!
+        if [ "$ARCH" = "aarch64" ]; then
+            echo "::group::Building custom patched Qemu"
+            "$DIR/BuildQemu.sh"
```
Comment (ADKaster) [inline Toolchain/BuildIt.sh:381]: Do we have to do this in BuildIt.sh, or can we do it in the CI script directly?

Comment (ADKaster) [reply Toolchain/BuildIt.sh:381]: eh. I guess a better question is if the BuildQemu.sh script smart enough to skip building things it doesn't have to when the "cached archive" is present.

Comment (BertalanD) [reply Toolchain/BuildIt.sh:381]: We could look into using `cache@v3` again and remove the cache handling logic from the build scripts.

Comment (BertalanD) [reply Toolchain/BuildIt.sh:381]: #19094

Comment (ADKaster) [reply Toolchain/BuildIt.sh:381]: Ah using that new set of cache steps, you could simply have a separate build step to build/cache qemu that only depends on BuildQemu.sh and our patches, and bypass doing it in BuildIt.sh altogether.

**Review by GMTA — APPROVED** [review]

**Review by ADKaster — APPROVED** [review]
Let's dew it

**Pull request merged**